cli/src/oxide_override.rs

```diff
@@ -0,0 +1,600 @@
+            })?;
+            let encoded_cert = base64::engine::general_purpose::STANDARD.encode(&cert_bytes);
+
+            *request = request.to_owned().body_map(|body| {
+                body.signing_keypair(DerEncodedKeyPair {
+                    private_key: encoded_privkey,
+                    public_cert: encoded_cert,
+                })
+            });
+        }
+        Ok(())
+    }
+
+    fn execute_networking_allow_list_update(
+        &self,
+        matches: &clap::ArgMatches,
+        request: &mut oxide::builder::NetworkingAllowListUpdate,
+    ) -> anyhow::Result<()> {
+        match matches
+            .get_one::<clap::Id>("allow-list")
+            .map(clap::Id::as_str)
+        {
+            Some("any") => {
+                let value = matches.get_one::<bool>("any").unwrap();
+                assert!(value);
+                *request = request
+                    .to_owned()
+                    .body_map(|body| body.allowed_ips(AllowedSourceIps::Any));
+            }
+            Some("ips") => {
+                let values: Vec<IpOrNet> = matches.get_many("ips").unwrap().cloned().collect();
+                *request = request.to_owned().body_map(|body| {
+                    body.allowed_ips(AllowedSourceIps::List(
+                        values.into_iter().map(IpOrNet::into_ip_net).collect(),
+                    ))
+                });
+            }
+            _ => unreachable!("invalid value for allow-list group"),
+        }
+
+        Ok(())
+    }
+}
+
+impl OxideOverride {
+    /// Construct a new OxideOverride for JSON output.
+    pub fn new_json() -> Self {
+        OxideOverride::Json {
+            needs_comma: AtomicBool::new(false),
+        }
+    }
+
+    /// Construct a new OxideOverride for tabular output.
+    pub fn new_table(fields: &[String]) -> Self {
+        OxideOverride::Table {
+            table: Box::new(Mutex::new(TableFormatter::new(fields))),
+        }
+    }
+
+    fn ip_range(matches: &clap::ArgMatches) -> anyhow::Result<IpRange> {
+        let first = matches.get_one::<IpAddr>("first").unwrap();
+        let last = matches.get_one::<IpAddr>("last").unwrap();
+
+        match (first, last) {
+            (IpAddr::V4(first), IpAddr::V4(last)) => {
+                let range = Ipv4Range::try_from(Ipv4Range::builder().first(*first).last(*last))?;
+                Ok(range.into())
+            }
+            (IpAddr::V6(first), IpAddr::V6(last)) => {
+                let range = Ipv6Range::try_from(Ipv6Range::builder().first(*first).last(*last))?;
+                Ok(range.into())
+            }
+            _ => anyhow::bail!(
+                "first and last must either both be ipv4 or ipv6 addresses".to_string()
+            ),
+        }
+    }
+}
+
+enum ReturnType {
+    Array,
+    Object,
+}
+
+/// Find the fields present on a the object returned by `success_item`.
+fn success_item_fields(root_schema: &RootSchema) -> (Vec<String>, ReturnType) {
```

Comment:
I'm concerned that this function seems fairly brittle. Is the idea that it will be mitigated by the tests?